core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -89,6 +92,9 @@ private[spark] object JettyUtils extends Logging {
             val result = servletParams.responder(request)
             response.setHeader("Cache-Control", "no-cache, no-store, must-revalidate")
             response.setHeader("X-Frame-Options", xFrameOptionsValue)
+            response.setHeader("X-XSS-Protection", xXssProtectionValue.get)
```

Comment:
This will fail if these aren't set. Only set the headers if the Option exists

Reply:
Yes, I will add a check for that.